### DISCUSSION TITLE
Understanding particle's Barcode

### DISCUSSION THREAD

**Discussion by pbelecky** [acts-project/acts#1798] [General]
Hello!

I'm trying to figure out how a Barcode for a particle is composed.
I read Barcode description from here:
https://github.com/acts-project/acts/blob/main/Fatras/include/ActsFatras/EventData/Barcode.hpp#L17

For example, suppose we have such particles:

                      primary point of interaction
                                  |
             --------------------------------------------
             |          |         |          |          |
         particle0  particle1  particle2  particle3  particle4
                                  |
                    -----------------------------
                    |        |        |         |                        
             particle5  particle6  particle7  particle8
                                      |
                        ------------------------------------
                        |           |           |          |
                    particle9  particle10  particle11  particle12

There were two colliding particle beams.
As a result of their collision, the following particles were appeared: particle0, particle1, particle2, particle3, particle4.
Particles particle5,  particle6,  particle7,  particle8 are obtained from particle2.
Particles particle9,  particle10,  particle11,  particle12 are obtained from particle7.
                    
What is the Barcodes of the following particles:
particle2, particle3,
particle7, particle8,
particle11, particle12 ?

**Comment by asalzburger** [comment]
Apologies, this has been totally under my radar.

The `Barcode.hpp` has quite some detailed information in the comments, the trick is that we have a certain bitset for particles, and then for generation, i.e. we encode the generation (the level in you tree structure) in the number of generation and then start counting the particles within a generation from 1 - N.

**Reply by paulgessinger** [reply]
cf #2041
